### PR TITLE
fix: added self-contained style for button-group component

### PR DESCRIPTION
**Pull request by JYSW380**
Closes SAP/fundamental-styles#145 

This PR contains the self-contained styling for the button-group component.
This PR is part of the larger task of making all components self-contained. #136

### CODE REVIEW

File: components/button-group.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+$fd-icons-path: "../scss/icons/";
+
+@import "../scss/components/button";
+@import "../scss/components/button-group";
+@import "../scss/icons/icon";
```
Comment (greg-a-smith) [inline components/button-group.scss]: This is another component that is a little strange to apply self-contained styles to because the majority of the style content loaded here is for the button component, which has its own self-contained styles. I know these were added because of the playground examples, but honestly the playground examples should be reworked to not include any specific fundamental-related content.  Although this is the button group component, the examples shouldn't use the button component for the children.  The essence of a button group is a wrapper around multiple elements (typically buttons although the CSS is written such that it wouldn't have to be) that have rounded corners and borders to make sure they format correctly when in a group.  The playground examples can show a `<button>` element, but shouldn't use any of the `fd-button` classes.  They could be styled with some custom CSS on the playground page itself using a temp class variable name (_e.g._ `btn-playground`).

On the main docs site, we can keep the buttons in place as the children as those are more use case-related.

Comment (JYSW380) [reply components/button-group.scss]: @greg-a-smith If I understand correctly, I should remove the `fd-button` class and add some custom class called `btn-playground` which has the same style as the fd-button? Where would I put this new class? Is there a file where we put custom sass in the docs? Or maybe just in `<style>` tags?

File: scss/components/button-group.scss [hunk 2]
```diff
@@ -2,6 +2,13 @@
 @import "./../mixins";
 @import "./../functions";
 
+
+.btn-playground {
```
Comment (droshev) [inline scss/components/button-group.scss]: I think this shouldn't be in that file

**Comment by droshev** [comment]
it would be great to add some values in the `ugly` theme for the btn group.
Also I don't see any constrains of having different than 3 buttons in a group, so it would be good to have examples with different set of buttons

File: components/button-group.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+$fd-icons-path: "../scss/icons/";
+
+@import "../scss/components/button-group";
+@import "../scss/icons/icon";
```
Comment (greg-a-smith) [inline components/button-group.scss]: Icons shouldn't be needed here.

File: test/templates/button-group/index.njk [hunk 4]
```diff
@@ -1,129 +1,74 @@
...
-          aria={ label: 'Chart', selected: true })
-        }}
-    {{-  button(
-          { icon: 'pool' },
-            modifier={ block: [size] },
-            aria={ label: 'Pool' })
-          }}
+    <button class="btn-playground sap-icon--survey" aria-label="Survey"></button>
+    <button class="btn-playground sap-icon--pie-chart" aria-label="Chart"></button>
+    <button class="btn-playground sap-icon--pool" aria-label="Pool"></button>
```
Comment (greg-a-smith) [inline test/templates/button-group/index.njk]: These buttons should _not_ be using icon classes from the style library.  These "plain" buttons do not need to look like SAP buttons or look like previous playground examples.  These should literally be very simple-styled buttons that exist only to verify button group styles (_e.g._  border radius on or off, borders not doubling, etc.)

File: test/templates/button-group/index.njk [hunk 5]
```diff
@@ -1,129 +1,74 @@
...
-          aria={ label: 'Chart' })
-        }}
-    {{-  button(
-          { icon: 'pool' },
-            modifier={ block: [""] },
-            aria={ label: 'Pool' })
-          }}
+    <button class="btn-playground sap-icon--survey" aria-label="Survey" disabled></button>
+    <button class="btn-playground sap-icon--pie-chart" aria-label="Chart"></button>
+    <button class="btn-playground sap-icon--pool" aria-label="Pool"></button>
```
Comment (greg-a-smith) [inline test/templates/button-group/index.njk]: Same comment as above about icons.

File: test/templates/button/component-anchor.njk [hunk 6]
```diff
@@ -5,6 +5,6 @@
 {%- macro anchor_button(properties={}, modifier={}, state={}, aria={}, classes="", size="") -%}
 {%- set is_default = modifier.block and modifier.block.length and not (modifier.block | join("")) %}
 
-<a class="{{ 'fd-button' if not modifier.block or modifier.block == "" or is_default }}{{ modifier.block | modifier('button') }}{{ ' fd-button--'+size if size }}{{ ' sap-icon--'+properties.icon if properties.icon }}{{ state | state }}{{ classes | classes }}"{{ aria | aria }}>{%- if properties.label %}{{properties.label}}{%- endif %}</a>
+<a class="{{ 'button' if not modifier.block or modifier.block == "" or is_default }}{{ modifier.block | modifier('button') }}{{ ' button--'+size if size }}{{ ' sap-icon--'+properties.icon if properties.icon }}{{ state | state }}{{ classes | classes }}"{{ aria | aria }}>{%- if properties.label %}{{properties.label}}{%- endif %}</a>
```
Comment (greg-a-smith) [inline test/templates/button/component-anchor.njk]: What's this change for?  Seems unrelated to this PR.

Comment (JYSW380) [reply test/templates/button/component-anchor.njk]: you are right, will go back to the original code, thanks.

File: test/templates/button-group/index.njk [hunk 7]
```diff
@@ -1,129 +1,44 @@
...
-        }}
-    {{-  button(
-          { label: 'Third' },
-            modifier={ block: [size] },
-            aria={ })
-          }}
-  {%- endcall %}
 
+<style>
+  .btn-playground {
```
Comment (jbadan) [inline test/templates/button-group/index.njk]: @greg-a-smith Can we move this to layout.css or some other shared css file. I think we have a lot of components coming up that need a placeholder button.

**Comment by greg-a-smith** [comment]
@jbadan @hertweckhr1 The `layout.css` actually isn't loaded on the self-contained pages.  That said, the point is still valid so I'll create a new CSS file that _IS_ loaded on self-contained pages that just contains some resusable styles for playground purposes.

File: test/resources/layout-shared.css [hunk 8]
```diff
@@ -0,0 +1,12 @@
+body {
+    background-color: rgb(243, 244, 245);
+    transition: background-color .25s;
```
Comment (greg-a-smith) [inline test/resources/layout-shared.css]: Added this part to shared just to make the self-contained frame easier to see.
